receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -480,6 +513,9 @@ func (c *tracesConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSe
 				return err
 			}
 
+			if c.headerExtractor != nil {
```

Comment:
I suggest using something like a `NopHeaderExtraction` that implements the same interface to save checking for nil and reducing the likelihood of a panic due to nil.

Reply:
Okay, will implement the interface and set `NopHeaderExtraction` as default.